benchmarks/benchmarks/anndata.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import annotations
+
+import tracemalloc
+
+import numpy as np
+
+from .utils import gen_adata
+
+
+class GarbargeCollectionSuite:
+    runs = 10
+
+    # https://github.com/pythonprofilers/memory_profiler/issues/402 and other backend does not pick this up
```

Comment:
Ok @ivirshup @flying-sheep  independent of this issue, were you aware that https://pypi.org/project/memory-profiler/ is a line-by-line profiler?  I think this is probably not very good for us, no?  When I read `get_peakmem` (i.e., the function in `utils` we wrote based on `memory_profiler`), I would think that it would really track the literal peak memory, and not the peak over individual line operations.

Reply:
Similarly: https://scverse.zulipchat.com/#narrow/stream/393966-scanpy-anndata-dev/topic/Benchmarking/near/442513634